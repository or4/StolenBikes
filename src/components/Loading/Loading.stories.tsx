import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text } from '@storybook/addon-knobs';

import { Loading } from './Loading';

const stories = storiesOf('Loading', module);
stories.addDecorator(withKnobs);

stories.add('Loading', () => {
    return (
        <Loading>
            <div>{getKnobs()}</div>
        </Loading>
    );
});

function getKnobs() {
    return text('Loading', 'Loading ...');
}
