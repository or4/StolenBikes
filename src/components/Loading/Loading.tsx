import React from 'react';
import styled from 'styled-components';

const Container = styled.div`
    margin-top: 40px;
`;

export const Loading: React.FC = () => <Container>Loading ...</Container>;
