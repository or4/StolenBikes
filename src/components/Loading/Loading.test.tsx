import React from 'react';
import { mount } from 'enzyme';

import { Loading } from '../Loading';

describe('Loading', () => {
    it('should render correctly', () => {
        const component = <Loading />;

        const container = mount(component);
        expect(container).toMatchSnapshot();
        container.unmount();
    });
});
