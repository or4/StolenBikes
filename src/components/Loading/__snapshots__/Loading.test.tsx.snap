// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render correctly 1`] = `
<Loading>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "bDWFJH",
            "rules": Array [],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bdVaJa bDWFJH"
      >
        Loading ...
      </div>
    </StyledComponent>
  </styled.div>
</Loading>
`;
